linker/jvm/src/main/scala/org/scalajs/linker/PathOutputDirectory.scala

```diff
@@ -100,6 +103,35 @@ object PathOutputDirectory {
     promise.future
   }
 
+  private def readFromChannel(chan: AsynchronousFileChannel): Future[ByteBuffer] = {
+    val size = blocking(chan.size())
+
+    if (size > Int.MaxValue)
+      throw new IOException(s"file is too large ($size bytes)")
+
+    val buf = ByteBuffer.allocate(size.toInt)
+
+    val promise = Promise[ByteBuffer]()
+
+    def readLoop(): Unit =
+      chan.read(buf, buf.position(), (), Handler)
+
+    object Handler extends CompletionHandler[Integer, Unit] {
+      def completed(read: Integer, unit: Unit): Unit = {
+        if (read == -1 || !buf.hasRemaining())
+          promise.success(buf.flip())
```

Comment:
This does not compile on JDK 8:
```
[error] /localhome/jenkins/sjsjenkins/workspace/scalajs-main_PR-4247@2/linker/jvm/src/main/scala/org/scalajs/linker/PathOutputDirectory.scala:122:35: type mismatch;
[error]  found   : java.nio.Buffer
[error]  required: java.nio.ByteBuffer
[error]           promise.success(buf.flip())
[error]                                   ^
[error] one error found
```
`flip()` did not yet have an override in `ByteBuffer` to return a `ByteBuffer`. Instead we receive the one from `Buffer` which returns a `Buffer`. Either cast it down:
```suggestion
          promise.success(buf.flip().asInstanceOf[ByteBuffer])
```
or use a block:
```suggestion
          promise.success({ buf.flip(); buf })
```